Support tencent cloud COS storage in `datafusion-cli` (#9734)
apache/arrow-datafusion: docs/source/user-guide/cli.md

Note that the `endpoint` format of oss needs to be: `https://{bucket}.{oss-region-endpoint}`

## Registering COS Data Sources

[Tencent cloud COS](https://cloud.tencent.com/product/cos) data sources can be registered by executing a `CREATE EXTERNAL TABLE` SQL statement.

```sql
CREATE EXTERNAL TABLE test
STORED AS PARQUET
OPTIONS(
    'aws.access_key_id' '******',
    'aws.secret_access_key' '******',
    'aws.cos.endpoint' 'https://cos.ap-singapore.myqcloud.com'
)
LOCATION 'cos://bucket/path/file.parquet';
```

The supported OPTIONS are:

- access_key_id
- secret_access_key
- endpoint

> Note that the `endpoint` format of oss needs to be: `https://cos.{cos-region-endpoint}`

Reviewer: cos instead of oss?

Author: No, just add cos external table doc.

Reviewer 2: ```suggestion
Note that the `endpoint` format of urls must be: `https://cos.{cos-region-endpoint}`
```

CREATE EXTERNAL TABLE test
STORED AS PARQUET
OPTIONS(

>     'gcp.service_account_path' '/tmp/gcs.json',

Reviewer: is it related to cos onboard?

Author: No, add the missing prefix gcp and aws to related external table options.
